views-core/src/main/java/io/micronaut/views/ViewsModelDecorator.java

```diff
@@ -17,20 +17,22 @@
 
 import io.micronaut.context.annotation.DefaultImplementation;
 import io.micronaut.core.annotation.NonNull;
-import io.micronaut.http.HttpRequest;
 
 /**
  * Decorates a Views Model.
+ * @param <T> The model type
+ * @param <R> The request type
  * @author Sergio del Amo
  * @since 3.0.0
  */
 @DefaultImplementation(DefaultViewsModelDecorator.class)
-public interface ViewsModelDecorator {
+@FunctionalInterface
+public interface ViewsModelDecorator<T, R> {
```

Comment:
document in the breaking changes the change to the `ViewsModelDecorator` making the parameters generics.